pkg/operator/encryption/shared_encryption.go

```diff
@@ -0,0 +1,867 @@
+// of a new key under the new mode even if encryptionSecretMigrationInterval has not been reached.
+const encryptionSecretMode = "encryption.apiserver.operator.openshift.io/mode"
+
+// encryptionSecretInternalReason is the annotation that denotes why a particular key
+// was created based on "internal" reasons (i.e. key minting controller decided a new
+// key was needed for some reason X).  It is tracked solely for the purposes of debugging.
+const encryptionSecretInternalReason = "encryption.apiserver.operator.openshift.io/internal-reason"
+
+// encryptionSecretExternalReason is the annotation that denotes why a particular key was created based on
+// "external" reasons (i.e. force key rotation for some reason Y).  It allows the key minting controller to
+// determine if a new key should be created even if encryptionSecretMigrationInterval has not been reached.
+const encryptionSecretExternalReason = "encryption.apiserver.operator.openshift.io/external-reason"
+
+// encryptionSecretFinalizer is a finalizer attached to all secrets generated
+// by the encryption controllers.  Its sole purpose is to prevent the accidental
+// deletion of secrets by enforcing a two phase delete.
+const encryptionSecretFinalizer = "encryption.apiserver.operator.openshift.io/deletion-protection"
+
+// encryptionSecretMigrationInterval determines how much time must pass after a key has been observed as
+// migrated before a new key is created by the key minting controller.  The new key's ID will be one
+// greater than the last key's ID (the first key has a key ID of 1).
+const encryptionSecretMigrationInterval = time.Hour * 24 * 7 // one week
+
+// In the data field of the secret API object, this (map) key is used to hold the actual encryption key
+// (i.e. for AES-CBC mode the value associated with this map key is 32 bytes of random noise).
+const encryptionSecretKeyData = "encryption.apiserver.operator.openshift.io-key"
+
+// These annotations try to scare anyone away from editing the encryption secrets.  It is trivial for
+// an external actor to break the invariants of the state machine and render the cluster unrecoverable.
+const (
+	kubernetesDescriptionKey        = "kubernetes.io/description"
+	kubernetesDescriptionScaryValue = `WARNING: DO NOT EDIT.
+Altering of the encryption secrets will render you cluster inaccessible.
+Catastrophic data loss can occur from the most minor changes.`
+)
+
+// encryptionConfSecret is the name of the final encryption config secret that is revisioned per apiserver rollout.
+// it also serves as the (map) key that is used to store the raw bytes of the final encryption config.
+const encryptionConfSecret = "encryption-config"
+
+// revisionLabel is used to find the current revision for a given API server.
+const revisionLabel = "revision"
+
+// these can only be used to encode and decode EncryptionConfiguration objects
+var (
+	encoder runtime.Encoder
+	decoder runtime.Decoder
+)
+
+func init() {
+	scheme := runtime.NewScheme()
```

Comment:
usually we make scheme and codecs global as well and only add the register calls here in init